File: src/megax/MegaApp/nextapp/lib/models/contact.model.ts
import * as yup from "yup";

// export interface Contact {
//   id: number;
//   name: string;
//   phone: string;
//   email: string;
//   address: string;
//   dob: Date | string | null;
//   relationship: string; //wife|child|relative
//   isPrimaryContact: boolean;
//   userId: number;
// }

export const contactSchema = yup.object({
  id: yup.number().nullable().default(0),
  userId: yup.number().nullable().default(null),
  name: yup.string().required(),
  phone: yup.string().nullable().default(null),
  email: yup.string().nullable().default(null),
  address: yup.string().nullable().default(null),
  dob: yup.string().nullable().default(null),
  relationship: yup.string().nullable().default(null),
  isPrimaryContact: yup.boolean().default(false),
});

export type Contact = yup.InferType<typeof contactSchema>;

File: src/megax/MegaApp.Core/Dtos/ContactModel.cs
using System.ComponentModel.DataAnnotations;

namespace MegaApp.Core.Dtos;

public record ContactModel
{
    public int Id { get; set; }

    [MaxLength(255)]
    [Required]
    public string Name { get; set; }

    [MaxLength(255)]
    public string Phone { get; set; }

    [MaxLength(255)]
    public string Email { get; set; }

    [MaxLength(255)]
    public string Address { get; set; }
    public DateTimeOffset? Dob { get; set; }

    [MaxLength(255)]
    public string Relationship { get; set; }//wife|child|relative
    public bool IsPrimaryContact { get; set; }

    public ContactModel()
    {

    }

    public ContactModel(Core.Db.Entities.Contact contact)
    {
        Id = contact.Id;
        Name = contact.Name;
        Phone = contact.Phone;
        Email = contact.Email;
        Address = contact.Address;
        Dob = contact.Dob;
        Relationship = contact.Relationship;
        IsPrimaryContact = contact.IsPrimaryContact;
    }
}
